frontend/src/Editor/Inspector/EventManager.jsx

```diff
@@ -672,38 +672,66 @@ export const EventManager = ({
-                              ?.value ?? param.defaultValue
-                          }
-                          onChange={(value) => {
-                            const newParam = { ...param, value: value };
-                            const params = event?.componentSpecificActionParams ?? [];
-                            const newParams = params.map((paramOfParamList) =>
-                              paramOfParamList.handle === param.handle ? newParam : paramOfParamList
-                            );
-                            handlerChanged(index, 'componentSpecificActionParams', newParams);
-                          }}
-                          enablePreview={true}
-                          type={param?.type}
-                          fieldMeta={{ options: param?.options }}
-                          cyLabel={param.displayName}
-                        />
-                      </div>
+                      {param.type === 'select' ? (
+                        <div className="col-9" data-cy="action-options-action-selection-field">
+                          <Select
+                            className={`${darkMode ? 'select-search-dark' : 'select-search'} w-100`}
+                            options={param.options}
+                            value={
+                              event?.componentSpecificActionParams?.find(
+                                (paramItem) => paramItem.handle === param.handle
+                              )?.value ?? param.defaultValue
+                            }
+                            search={true}
+                            onChange={(value) => {
+                              const newParam = { ...param, value: value };
+                              const params = event?.componentSpecificActionParams ?? [];
+                              const newParams = params.map((paramOfParamList) =>
+                                paramOfParamList.handle === param.handle ? newParam : paramOfParamList
+                              );
+                              handlerChanged(index, 'componentSpecificActionParams', newParams);
+                            }}
+                            placeholder={t('globals.select', 'Select') + '...'}
+                            styles={styles}
+                            useMenuPortal={false}
+                            useCustomStyles={true}
+                          />
+                        </div>
+                      ) : (
+                        <div
+                          className={`${
+                            param?.type
+                              ? param.type == 'select'
+                                ? 'col'
+                                : 'col-7'
+                              : 'col-9 fx-container-eventmanager-code'
+                          } fx-container-eventmanager ${param.type == 'select' && 'component-action-select'}`}
```

Comment:
Can you remove the `param.type == 'select'` check? This will not be executed as we already have a check at the top